compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -29,6 +30,39 @@ use provenance_map::*;
 
 pub use init_mask::{InitChunk, InitChunkIter};
 
+/// Functionality required for the bytes of an `Allocation`.
+pub trait AllocBytes:
+    Clone + fmt::Debug + Eq + PartialEq + Hash + Deref<Target = [u8]> + DerefMut<Target = [u8]>
+{
+    /// Adjust the bytes to the specified alignment -- by default, this is a no-op.
+    fn adjust_to_align(self, _align: Align) -> Self;
```

Comment:
That's a strange API. Shouldn't alignment be set when the `Self` is created?

Reply:
This is called in `adjust_from_tcx`, it would be necessary if it could accept another bytes type that may not be aligned, but as is I think you're right - we always create the item with the desired alignment and don't need to re-align it later.

Reply:
`adjust_from_tcx` will always have to convert a `Box<[u8]>` into an aligned byte storage, so that is probably the API we want in the trait.